fix(ddm): add to dashboard modal

obostjancic opened the pull request:
- Fixes #65426 

Before:
<img width="620" alt="image" src="https://github.com/getsentry/sentry/assets/86684834/e9d433ae-f005-457f-9ebf-0fd353fa18c6">

After:
<img width="620" alt="image" src="https://github.com/getsentry/sentry/assets/86684834/aade23d6-8b59-413a-8d90-63c66fa955ab">

k-fish committed:
clean(metrics-extraction): Remove hashes mismatch message (#65419)

### Summary
We already have a metric for this (`failed_on_demand_hashes`), which we
can alert on. Now that we're no longer developing we don't need to
check. This message can perhaps be moved to the widget save in the
future if we need debugging as that should also show drift in the metric
hashes.

obostjancic committed:
ref(ddm): streamline metrics query type (#65420)

obostjancic committed:
fix(ddm): add to dashboard modal

obostjancic committed:
height fix

gggritso committed:
feat(perf): Improve chart error handling in new modules (#65364)

A few improvements to module charts!

1. A small refactor. Changes our chart component to accept an `error`
object instead of booleans like `errored` or `isError`. This is a little
simpler, since it accepts all error-like objects, but also is more
powerful since we can, if we want, use the error info to render some
error text.
2. A small improvement. Adds error states to the charts in the Database
and HTTP modules

szokeasaurusrex committed:
ref: Delete unnecessary `start_transaction` (#65423)

This `start_transaction` call is unnecessary and leads to undefined
behavior. This function is a Celery task, and therefore the Sentry SDK's
Celery integration automatically instruments it.

This `start_transaction` call is the final one we intend to remove in
#63590; therefore, merging this PR closes #63590.

Abdkhan14 committed:
feat(traceview): Added warning handling for empty trace type. (#65424)

We were throwing invalid trace type error for empty traces.

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

ArthurKnaus commented:
This PR seems to be in a loop of applying `pre-commit` and eslint style` fixes...

ArthurKnaus commented:
Canceled linting check to stop the endless loop

wedamija committed:
feat(crons-db): Use `environment_id` when creating `MonitorEnvironment` models (#65263)

We want to move all crons related tables to a separate database. This
means we need to stop referring to foreign key relations in Django and
just use ids on creation.

This pr changes uses of `environment=environment` to
`environment_id=environment.id` so that we're not relying on the fk.

Zylphrex committed:
feat(metrics): Switch to new endpoint for metric samples (#65434)

This switches from the events endpoint over to the new metrics samples
endpoint to find samples that are better distributed over the time
range.

roggenkemper committed:
chore(inbound-filters): Reorder legacy browser filters (#65343)

this pr reorders the legacy browser filters. instead of having all the
old ones then the new ones, now they are ordered by browser. They are
mostly ordered by popularity, with Edge being moved lower to be near IE.

JonasBa committed:
feat(trace): trim span descriptions (#65431)

Trim long span descriptions to 100 characters

JonasBa committed:
feat(trace): adjust space depending on txn children timings (#65428)

We were only considering timings from the first txn children, we need to
consider it for all nodes in the tree.

obostjancic commented:
> Canceled linting check to stop the endless loop

Thanks!

schew2381 committed:
fix(integration): Have factory method match code change (#65360)

JonasBa committed:
feat(trace): do not inject missing spans for javascript platform (#65435)

We'll do this later based off the SDK name so that it is more accurate,
but the change will be trivial to make.

markstory committed:
fix(integrations) Don't pass orgid to features.has() (#65433)

We need an Organization shaped object not an integer to do feature flag
checks.

Fixes SENTRY-2NQR

gggritso committed:
feat(perf): Improve table error handling in HTTP and Database modules (#65430)

Luckily, `GridEditable` already supports the `error` prop, so all I had
to do was pass it down from the API response.

Abdkhan14 committed:
fix(new-trace-view): Incorrect index for adding adding children  to s… (#65438)

Fixes:
![Screenshot 2024-02-20 at 12 44 59
PM](https://github.com/getsentry/sentry/assets/60121741/5a3fc256-13f6-4382-8e33-fb390dc28a8d)

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

JoshFerge committed:
fix(replays): correctly use ip_address field instead of ip (#65439)

Fixes SENTRY-17QY

TODO: follow up with a test / runtime behavior that confirms validation
like we do for feedbacks. for now just trying to clean up the issues
feed for this.

jangjodi committed:
ref(similarity-embedding): Add would group to bulk analytics (#65323)

Add whether the similarity algorithm would've grouped the issues to the
bulk agree/disagree analytics

ykamo001 committed:
fix(integration): Prevent Sentry issue from raising (#65436)

We were raising ungrouped errors on this flow due to the error message,
and the side effect was that we blocked continuous deployment. In
reality, this is an expected error, and we want to just show the user
the error message. Switching the type of the exception helps prevent
this due to the way the Pipeline flow works.

JoshFerge committed:
fix(issues): make user object allow additional properties (#65440)

A new field, `sentry_user` was added to relay yesterday. We need to
allow additional properties in our user validation schema in order to
allow events with this through. Test below fails on master and is fixed
here.

https://github.com/getsentry/relay/pull/3122

asottile-sentry committed:
ref: upgrade coverage (#65432)

the latest version has some specialization for 3.12 that I'd like to try
out

<!-- Describe your PR here. -->

JoshFerge committed:
fix(replays): fix rage click option in project serializer (#65353)

- [x] Don't set a default value in the .get here, as we already set a
default here which should be obeyed:
https://github.com/getsentry/sentry/blob/e3629860197903f263c6bcdf6fac37c7d7a0a302/src/sentry/projectoptions/defaults.py#L135
- [x] don't compare it to "1", as the value is a boolean in the database
/ default.

Observed an issue where if you set it to false, then back to true, it
would appear in the UI as if it were still false.

sidenote: we desperately need to refactor these project options, it is
quite footgun-prone now.

joshuarli committed:
devenv: allow starting colima on macos-13 for ci (#65373)

JonasBa committed:
feat(trace): animate row into view after user finishes scrolling (#65422)

When a user finishes scrolling on X axis and no rows are visible as they
are either to the left or right of the view, scroll the list back into
place.

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

azaslavsky committed:
test(backup): Add release test at 24.2.0 (#65450)

This commit also removes the tests at 23.12.0, as those are now outside
the support window.

k-fish committed:
fix(metrics-extraction): Always emit on-demand for widget (#65449)

### Summary
On-demand was being conditionally emitted depending on the
supports_metrics state before. The continue can be hit by the regular
support_metrics flow, and there are existing on-demand rows that aren't
appearing for some orgs, so this should show them if they qualify from
'get_all_widgets_of_organization'

narsaynorath committed:
feat(app-start): Update charts on overview page (#65441)

Adds two new charts for rendering the average duration comparison
between two releases as well as the count of start events for each
release. The count widget gives context regarding the amount of adoption
of the releases you're comparing.

The middle chart is the Top Screens chart (in this example we only have
one, but it goes up to 5) which shows the start times of each screen
individually.

Zylphrex committed:
tests(metrics): Decrease flakiness of metrics samples test (#65456)

There seems to be some differences between how the timestamps are stored
in the 2 datasets that rounds differently. For now, just ensure we
truncate to the nearest second to stability. Eventually, all the queries
will be done on the same dataset which will eliminate this problem.

Fixes SENTRY-TESTS-ZGE

markstory committed:
feat(services) Add a factory for making delegators (#65342)

When doing the rc-processing cutover we had a few stumbles in getting
the splitter logic correct. We shouldn't have to re-invent this logic
each time we need to move infra around. This should help with future
cutover work.

snigdhas committed:
fix(first-release): Revise logic for firstRelease with no environments (#65361)

Fix a bug in how we query groups releases when no environment is
provided

gggritso committed:
fix(perf): Remove aggregate selector from HTTP module duration chart (#65429)

Also temporary, but even temporarily it doesn't make sense to use the
same `DurationChart` as the database module, since that one has a
percentile selector which uses a database-specific feature flag. So,
adding a `DurationChart` for the HTTP module.

k-fish committed:
fix(metrics-extraction): Improve on-demand check in the UI (#65459)

### Summary
Once the feature flag is enabled, we want to start sending 'useOnDemand'
to the backend for those users. On demand is able to support almost
everything now, so we can remove the majority of these conditions.

The rest are handled on the backend and fallback. There is a few small
exceptions in which we should not send the useOnDemand, primarily error
conditions, and ambiguous conditions like message or free-text.

IanWoodard committed:
chore(ci): Updating codecov-action to fix deprecation warning (#65367)

Bumping codecov-action version in order to move off of node16 which is
now deprecated.

iamrajjoshi committed:
feat(sidebar): Removed superuser warning banner for demo org (#65283)

Removed the warning banner for superuser for the demo org

Before:
<img width="1797" alt="Screenshot 2024-02-15 at 1 38 47 PM"
src="https://github.com/getsentry/sentry/assets/33237075/ca591100-2b5a-431f-a014-d71ce3cafe24">

After:
<img width="1797" alt="Screenshot 2024-02-15 at 1 31 40 PM"
src="https://github.com/getsentry/sentry/assets/33237075/0791ccca-24ad-45c7-8368-5812993e5464">

Fixes
https://github.com/getsentry/team-core-product-foundations/issues/73

schew2381 committed:
chore(staff): Allow staff to access org projects endpoint (#65404)

cathteng committed:
chore(team-org-roles): remove feature-flagged ability to set team org role (#65341)

malwilley committed:
ref(issue-stream): Refactor issue stream action logic with better typing and flexibility (#65443)

Before adding priority to the list of issue actions that remove issues
for the stream (https://github.com/getsentry/sentry/issues/65212), I
wanted to do some cleanup since there is a lot of existing logic here
that can be simplified and typed better.

Some of the changes: 

- The "For Review" tab and other issue actions were using separate code
paths, I've combined them. This means that `onMarkReviewed` was removed,
and the `removeItems` state which was only used by that code path was
also removed.
- Removed the `actionTakenGroupData` state. Instead of storing that in
state, we send it in through the `onActionTaken` callback.

joshuarli committed:
devenv: add repo-local sync (#65453)

snigdhas committed:
feat(issue-priority): Add support for issue.priority search (#65444)

Undo revert of https://github.com/getsentry/sentry/pull/65205
The previous PR had incorrectly setup tests in `test_activity.py`.
Refreshing the group from the db fixed the types and raised the expected
error, which is fixed in this PR.

Adds support for searching for issue.priority with a single value or an
array of values.


![image](https://github.com/getsentry/sentry/assets/16563948/1ffdfc68-d7c2-4c5a-a01a-2d376d46dc67)

Fixes https://github.com/getsentry/sentry/issues/65097

MichaelSun48 committed:
ref(sentry-answers): Refactor errorConfig to check for correct platform and add new sentry answers  (#65337)

Described in #64889

- **Refactored errorConfig to check for correct platform** (prev line
252 in errorConfig.tsx)
prev: we only checked if the issue was a nextJs issue
now: all sentry answers issues are mapped to their correct platform(s),
and we check to make sure both the issue title and platform match what
we expect

- **Added 6 new "sentry answers", listed in this
[issue](https://github.com/getsentry/sentry/issues/64889)**
Screenshots of 5 of them: (couldn't reproduce the nextRouter one)
<img width="952" alt="Pasted Graphic 5"
src="https://github.com/getsentry/sentry/assets/55160142/161a35cc-bc29-4592-befe-fc67e8566304">
<img width="962" alt="Pasted Graphic 9"
src="https://github.com/getsentry/sentry/assets/55160142/88784548-f3dd-485c-8a3f-75e8c6a8eb97">
<img width="960" alt="Pasted Graphic 7"
src="https://github.com/getsentry/sentry/assets/55160142/b1c1d5ea-479c-457e-abef-96eadaa2d7b6">
![Pasted Graphic
2](https://github.com/getsentry/sentry/assets/55160142/76cdc719-79c9-458c-8af9-819f36859358)
<img width="956" alt="Pasted Graphic 8"
src="https://github.com/getsentry/sentry/assets/55160142/397d06e9-ae56-4361-ac23-1f1d6f4470e2">

---------

Co-authored-by: Michael Sun <michaelsun@K0FJYPQJ6L.local>

michellewzhang committed:
ref(tsc): convert accountSecurity index to FC (#65303)

ref https://github.com/getsentry/frontend-tsc/issues/2

- converts this file into FC
- didn't convert to `useApiQuery` here (wasn't sure if that was
necessary?) - still using `await api.requestPromise`

gggritso committed:
fix(perf): Fix bad wrapping of Query module no data message (#65466)

It needs a grid wrapper, or it doesn't expand to full width! I forgot to
wrap it in #65365. Closes #65458.

wmak committed:
experiment(trace): Add an augment only flag (#65469)

- Lets try running the augment function only, if this still causes
timeouts the issue is with the augmentation. If this doesn't then the
issue lies in the new serialize functionality

wedamija committed:
feat(crons-db): Stop accessing `MonitorEnvironment.environment` directly (#65305)

We want to move all crons related tables to a separate database. To
facilitate this, we need to stop accessing the
`MonitorEnvironment.environment` relation, and use a helper function
instead.

wedamija committed:
feat(crons-db): Add caching when we fetch environment (#65475)

Missed this in https://github.com/getsentry/sentry/pull/65305. We want
to cache envs if possible.

wedamija committed:
feat(crons-db): Stop querying `MonitorEnvironment.environment` in queries (#65368)

We want to move all crons related tables to a separate database. To
facilitate this, we need to stop filtering to Environment using a join
and instead fetch ids from Environment in a separate query.

leeandher committed:
fix(commit-context): Update activity text (#65473)

Before:

<img width="544" alt="image"
src="https://github.com/getsentry/sentry/assets/35509934/a5e60091-8392-4890-88b9-23017a75503b">


After:
<img width="405" alt="image"
src="https://github.com/getsentry/sentry/assets/35509934/86571ce5-7e5c-44e7-bcf5-134fa7b2bb22">

michellewzhang committed:
fix(feedback): check and filter if contexts feedback message is none (#65479)

Fixes SENTRY-2NH5

- check if a feedback is missing `contexts`, `contexts[feedback]`, or
`contexts[feedback][message]`
- if so, filter it out first

anonrig committed:
build: disable biome formatter for now (#65464)

michellewzhang committed:
ref(replays): update user action breadcrumb color (#65485)

Per this
[figma](https://www.figma.com/file/uYS5xCq0m3SjtLKlxl7ilC/Specs%3A-Component-Names-in-Replay-Detail?type=design&node-id=0-1&mode=design&t=boX77bWp0pvCtnAh-0),
we're changing all the user action breadcrumb items to be blue in
preparation for react component names:

<img width="583" alt="SCR-20240220-nsto"
src="https://github.com/getsentry/sentry/assets/56095982/17477bc2-c1ef-4a15-ba12-a0e51c513cba">
<img width="588" alt="SCR-20240220-nrcv"
src="https://github.com/getsentry/sentry/assets/56095982/f5835bbd-bcb0-4065-b0ef-516209a9ca4f">

cathteng committed:
chore(slack): log entire slack unfurl error (#65471)

Abdkhan14 committed:
fix(trace-view-warnings): Added new svg. (#65487)

Changed image for banner:
<img width="1502" alt="Screenshot 2024-02-20 at 6 42 36 PM"
src="https://github.com/getsentry/sentry/assets/60121741/5b0a00b5-681e-4400-a736-80ff83568367">

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

wedamija committed:
feat(crons-db): Break foreign key on `MonitorEnvironment.environment` (#65370)

This removes the fk from `MonitorEnvironment.environment` and the
relationship from Django

JoshFerge committed:
fix(issues): allow additional properties on issues platform event schema (#65493)

set additionalProperties to true on the issue platform JSON validation
schema to be more lax, which should help with upstream changes where
more fields are added to events.

Related to #inc-646

mrduncan committed:
ref(issues): Mark team endpoints private (#65468)

These are only used on a single table within the stats page and will
potentially be changed in the not too distant future, also confirmed all
have extremely low usage.

scttcper committed:
chore(ui): Fix formatting error in releaseChart (#65497)

Commit:
chore(grouping): fix typo (#65503)

Dav1dde committed:
feat(relay): Add bucket encoding global option (#65410)

Commit:
feat(ddm): Add instrumentation for query timerange (#65425)

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes
